src/main/java/com/boardgame/controllers/GameController.java

```diff
@@ -18,16 +17,20 @@
 @RestController
 public class GameController {
 
-	@Autowired private GameRepository gameRepo;
-	@Autowired private UserRepository userRepo;
+	@Autowired
+	protected GameService gameService;
+
 
 	@RequestMapping("/game/{name}/create")
-	public @ResponseBody Game createGame(@PathVariable("name") String name){
-		if(gameRepo.findByName(name) != null)
-			throw new BadRequestException(String.format("Game name '%s' already exist.", name));
-		return gameRepo.save(new Game(name));
+	public @ResponseBody
+	Game createGame(@PathVariable("name") String name, @RequestBody Map<String, Object> rules){

```

Comment:
current rules param to be changed to refernece in registerGames (see bellow)

register Game will be an authenticated call in a production environment to load the compilation of javascript into the execution environment and compiled by rhino (or node) when needed.

Register game will create a document GameRules
GameRules will have the needed pieces to define the game and run games.